sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -392,7 +392,9 @@ case class InsertIntoHiveTable(
 
     // Invalidate the cache.
     sqlContext.sharedState.cacheManager.invalidateCache(table)
-    sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    if (partition.nonEmpty) {
+      sqlContext.sessionState.catalog.refreshTable(table.catalogTable.identifier)
+    }
```

Comment:
let's revert it first, we should think about cache and refresh more thorough later.